src/core/server/http/http_tools.ts

```diff
@@ -79,11 +79,30 @@ export function getServerOptions(config: HttpConfig, { configureTLS = true } = {
   return options;
 }
 
-export function createServer(options: ServerOptions) {
-  const server = new Server(options);
+export function getListenerOptions(config: HttpConfig) {
+  return {
+    keepaliveTimeout: config.keepaliveTimeout,
+    socketTimeout: config.socketTimeout,
+  };
+}
+
+interface ListenerOptions {
+  keepaliveTimeout: number;
+  socketTimeout: number;
+}
+
+export function createServer(serverOptions: ServerOptions, listenerOptions: ListenerOptions) {
+  const server = new Server(serverOptions);
 
-  // Revert to previous 120 seconds keep-alive timeout in Node < 8.
-  server.listener.keepAliveTimeout = 120e3;
+  server.listener.keepAliveTimeout = listenerOptions.keepaliveTimeout;
+  server.listener.setTimeout(listenerOptions.socketTimeout);
+  server.listener.on('timeout', socket => {
+    if (socket.writable) {
+      socket.end(Buffer.from('HTTP/1.1 408 Request Timeout\r\n\r\n', 'ascii'));
```

Comment:
adding a test for timeout shouldn't be a problem. in http_server.test.ts:
```js
const delay = (ms: number) => new Promise(res => setTimeout(res, ms));
test('returns 408 on timeout error', async () => {
  const router = new Router('');

  router.get({ path: '/', validate: false }, async (req, res) => {
    await delay(1000);
    return res.ok({});
  });

  const { registerRouter, server: innerServer } = await server.setup({
    ...config,
    socketTimeout: 1,
  });
  registerRouter(router);

  await server.start();

  await supertest(innerServer.listener).get('/').expect(408)
});
```

Reply:
added!